extensions/MESA/EGL_MESA_config_select_group.txt

```diff
@@ -0,0 +1,108 @@
+Name
+
+    EGL_MESA_config_select_group
+
+Name Strings
+
+    EGL_MESA_config_select_group
+
+Contributors
+
+    Hal Gentz <zegentzy@protonmail.com>
+    Adam Jackson <ajax@redhat.com>
+    Robert Mader <robert.mader@posteo.de>
+
+Contacts
+
+    Hal Gentz <zegentzy@protonmail.com>
+
+Status
+
+    Draft
+
+Version
+
+    Version 3, 2020-04-06
+
+Number
+
+    EGL Extension #143
+
+Extension Type
+
+    EGL client extension
+
+Dependencies
+
+    This extension is written against the wording of the 2014.08.27 revision
+    of the EGL 1.5 Specification.
+
+Overview
+
+    This extension provides a mechanism to lower the config selection priority
+    of the configs returned by eglChooseConfig. A new config attribute is
+    introduced, providing a way for the implementation to group configs into
+    different config selection categories.  The config selection priorities of
+    all the configs in one category may be higher or lower than the ones in
+    other categories.  The config selection priorities in one category follow
+    the rules of eglChooseConfig.
+
+New Types
+
+    None
+
+New Procedures and Functions
+
+    None
+
+New Tokens
+
+    Accepted as a new EGLConfig attribute:
+
+        EGL_CONFIG_SELECT_GROUP_MESA                0x31DF
+
+Additions to the EGL 1.5 Specification
+
+    In section 3.4 "Configuration Management", under the "Buffer Descriptions
+    and Attributes" heading, append to table 3.1 the following:
+
+    "EGL_CONFIG_SELECT_GROUP_MESA | integer | the config select group"
+
+    In section 3.4 "Configuration Management", under the "Other EGLConfig
+    Attribute Descriptions" heading, append to the end the following:
+
+    "The attribute EGL_CONFIG_SELECT_GROUP_MESA is used to specify the config
+    selection category of the config. Configs in the i-th category have a higher
+    config selection priority than those found in the (i+1)-th category.
+
+    "The attribute EGL_CONFIG_SELECT_GROUP_MESA is used by the implementation to
+    override the default sorting rules, by collecting configurations into
+    groups. It is not accepted by the <attrib_list> parameter of
+    eglChooseConfig, but it is accepted as the <attribute> in
+    eglGetConfigAtrrib."
+
+    In section 3.4 "Configuration Management", under the "Sorting of EGLConfigs"
+    heading, append to table 3.4 the following:
+
+    "EGL_CONFIG_SELECT_GROUP_MESA | 0 | Ignore | Smaller | 0"
+
+    In section 3.4 "Configuration Management", under the "Sorting of EGLConfigs"
+    heading, append before the list entry which starts with "1. Special: by
+    EGL_CONFIG_CAVEAT" the following entry:
+
+    "0. Smaller EGL_CONFIG_SELECT_GROUP_MESA."
```

Comment:
Is sorting by group always enabled? or what is the default behavior, and when is new sort rule 0 enabled?
This would override current sorting rules and give app different results than prior to this. This should be a clear opt-in case for the app.

Reply:
Not to dissuade you from your design, But have you thought about co-opting the EGL_NATIVE_VISUAL_TYPE attribute for a X11/MESA special rule - exact match for anything greater than EGL_DONT_CARE. Then if attribute_list has EGL_NATIVE_VISUAL_TYPE, EGL_CONFIG_TRANSPARENT_MESA (new token). Then only those configs would be returned - same sorting order. Selection rule being "Exact" match, sorting none?. This attribute is limited to windows surfaces (section 3.4.1.1). Maybe that gets you what you want?

Reply:
`EGL_NATIVE_VISUAL_TYPE` is already used to communicate the difference between TrueColor and PseudoColor (or StaticColor or DirectColor or...) visuals in X11. The "transparent" visual under X is just one whose depth exceeds the number of r/g/b bits.